python_modules/dagster/dagster/core/decorators.py

```diff
@@ -47,33 +72,46 @@ def __call__(self, fn):
             self.name = fn.__name__
 
         _validate_transform_fn(self.name, fn, self.inputs, expect_context)
-        transform_fn = _create_transform_wrapper(fn, self.inputs, expect_context)
-        return SolidDefinition.single_output_transform(
+        transform_fn = _create_transform_wrapper(fn, self.inputs, self.outputs, expect_context)
+        return SolidDefinition(
             name=self.name,
             inputs=self.inputs,
-            output=self.output,
+            outputs=self.outputs,
             transform_fn=transform_fn,
+            config_def=self.config_def,
             description=self.description,
         )
 
 
-def solid(*, name=None, inputs=None, output=None, description=None):
-    return _Solid(name=name, inputs=inputs, output=output, description=description)
+def solid(*, name=None, inputs=None, output=None, outputs=None, description=None):
+    return _Solid(name=name, inputs=inputs, output=output, outputs=outputs, description=description)
 
 
-def _create_transform_wrapper(fn, inputs, include_context=False):
+def _create_transform_wrapper(fn, inputs, outputs, include_context=False):
     input_names = [input.name for input in inputs]
 
     @wraps(fn)
-    def transform(context, args):
+    def transform(context, args, config):
         kwargs = {}
         for input_name in input_names:
             kwargs[input_name] = args[input_name]
 
         if include_context:
-            return fn(context, **kwargs)
+            result = fn(context, **kwargs)
+        else:
+            result = fn(**kwargs)
+        if inspect.isgenerator(result):
+            yield from result
         else:
-            return fn(**kwargs)
+            if isinstance(result, Result):
```

Comment:
This does end up being quite lolzy, but it's the price we pay for the magic python API.